### PR TITLE
PR Preview: Add Preview Pull Request Metrics

### PR DESCRIPTION
**Pull request by tidy-dev**
## Description
This PR Adds the following metric tracking:
- `previewedPullRequestCount`: This is to see how many users are using the feature.
- `createPullRequestFromPreviewCount`: This is to see how many users are creating pull requests on dotcom from the feature. The existing metric of `createPullRequestCount` which will track all times this occurs.
   - Answer: Will users use this or just go back to always going straight to dotcom? 
  

## Release notes
Notes: no-notes

### CODE REVIEW

**Review by IcebergVerse — COMMENTED** [review]
add-preview-pull-request-metrics

File: app/src/lib/stats/stats-store.ts [hunk 1]
```diff
@@ -1981,6 +1993,15 @@ export class StatsStore implements IStatsStore {
       log.error(`Error reporting opt ${direction}:`, e)
     }
   }
+
+  /**
+   * Increments the `createPullRequestFromPreviewCount` metric
```
Comment (sergiou87) [inline app/src/lib/stats/stats-store.ts]:
```suggestion
   * Increments the `previewedPullRequestCount` metric
```

File: app/src/ui/open-pull-request/open-pull-request-dialog.tsx [hunk 2]
```diff
@@ -83,8 +83,8 @@ export class OpenPullRequestDialog extends React.Component<IOpenPullRequestDialo
       dispatcher.showPullRequest(repository)
     } else {
       dispatcher.createPullRequest(repository)
-      // TODO: create pr from dialog pr stat?
       dispatcher.recordCreatePullRequest()
+      dispatcher.recordCreatePullRequestFromPreview()
```
Comment (sergiou87) [inline app/src/ui/open-pull-request/open-pull-request-dialog.tsx:88]: It'll be important to mention _somewhere_ that these two metrics will be tracked at the same time in this case.

Comment (tidy-dev) [reply app/src/ui/open-pull-request/open-pull-request-dialog.tsx:88]: I added it to the `createPullRequest` metric documentation. To me if I when looking for "how many times do users create a pull request on dotcom", I would expect `createPullRequestCount` to answer that question. But, I can see that if I also so the metric below it, I could wonder if they were included in that count.. Thus, I think this is the most logical place to note it.

**Review by sergiou87 — DISMISSED** [review]
Thanks for those changes! 🙌

**Review by sergiou87 — DISMISSED** [review]

**Review by sergiou87 — APPROVED** [review]

**Pull request merged**